### ISSUE TITLE
dart2js Exception

### ISSUE THREAD

**Issue by c-antin** [peiffer-innovations/rest_client#4]
I encountered the following error when building for web (`flutter build web`):

_Target dart2js failed: Exception: /C:/tools/pub/cache/hosted/pub.dartlang.org/rest_client-2.0.0/lib/src/client/client.dart:158:44:
Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.
          responseBody = await processJson(body);
                                           ^
Error: Compilation failed._

Possible fix:
`responseBody = await processJson(body!);`

**Comment by jpeiffer** [comment]
Hmmm...  It works for me on mobile.  I'll try to look into the web side this week.  Thanks for the report.

**Comment by jpeiffer** [comment]
In the mean time, if you have a simple example that reproduces this, please provide it.  That'd be helpful for speeding up the resolution.

**Comment by c-antin** [comment]
Here is an example `main.dart`: 
```dart
import 'package:json_dynamic_widget/json_dynamic_widget.dart';

void main() {
  JsonWidgetRegistry.instance;
}
```
`pubspec.yaml`
```yaml
environment:
  sdk: ">=2.12.0 <3.0.0"
```
With the latest flutter stable release the error still occurs when building for web.

Fix: adding postfix null assertion “bang” operator (!), since the boolean expression never allows `body` to be `null`:
```dart
        dynamic responseBody;
        if (body?.isNotEmpty == true) {
          responseBody = await processJson(body!);
        }
```

**Comment by jpeiffer** [comment]
Thanks!  Checked in a fix.  Should be resolved with 2.0.1.
